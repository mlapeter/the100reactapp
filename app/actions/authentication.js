export const FETCH_TOKEN = "FETCH_TOKEN";
export const FETCH_TOKEN_RESULT = "FETCH_TOKEN_RESULT";
export const FETCH_TOKEN_ERROR = "FETCH_TOKEN_ERROR";

export const DECODE_TOKEN = "DECODE_TOKEN";
export const DECODE_TOKEN_RESULT = "DECODE_TOKEN_RESULT";
export const DECODE_TOKEN_ERROR = "DECODE_TOKEN_ERROR";
export const REMOVE_TOKEN = "REMOVE_TOKEN";
export const REMOVE_TOKEN_ERROR = "REMOVE_TOKEN_ERROR";
export const SET_FIREBASE_TOKEN = "SET_FIREBASE_TOKEN";
export const RESET_PASSWORD = "RESET_PASSWORD";
export const RESET_PASSWORD_RESULT = "RESET_PASSWORD_RESULT";
export const RESET_PASSWORD_ERROR = "RESET_PASSWORD_ERROR";

export const fetchToken = (username, password, temp_auth_token) => ({
  type: FETCH_TOKEN,
  username,
  password,
  temp_auth_token
});

export const decodeToken = token => ({
  type: DECODE_TOKEN,
  token
});

export const removeToken = () => ({
  type: REMOVE_TOKEN
});

export const setFirebaseToken = token => ({
  type: SET_FIREBASE_TOKEN,
  token
});

export const resetPassword = resetPasswordEmail => ({
  type: RESET_PASSWORD,
  resetPasswordEmail
});
