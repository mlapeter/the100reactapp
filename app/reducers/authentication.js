import {
  FETCH_TOKEN,
  FETCH_TOKEN_RESULT,
  FETCH_TOKEN_ERROR,
  DECODE_TOKEN,
  DECODE_TOKEN_RESULT,
  DECODE_TOKEN_ERROR,
  REMOVE_TOKEN,
  REMOVE_TOKEN_ERROR,
  SET_FIREBASE_TOKEN
} from "../actions/authentication";

const initialState = {
  appLoading: true,
  isLoading: false,
  isAuthed: false,
  user: {},
  token: "",
  firebaseToken: ""
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TOKEN:
      return {
        ...state,
        username: action.username,
        password: action.password,
        isLoading: true
      };
    case FETCH_TOKEN_RESULT:
      return {
        ...state,
        token: action.token,
        firebaseToken: action.firebaseToken
      };
    case FETCH_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        token: "",
        firebaseToken: ""
      };
    case DECODE_TOKEN:
      return {
        ...state,
        token: action.token,
        isLoading: true
      };
    case DECODE_TOKEN_RESULT:
      return {
        ...state,
        user: action.result,
        isLoading: false,
        isAuthed: true
      };
    case DECODE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false
      };
    case REMOVE_TOKEN:
      return {
        ...state,
        isAuthed: false,
        isLoading: false,
        user: {},
        token: "",
        firebaseToken: ""
      };
    case REMOVE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        user: {},
        token: ""
      };
    case SET_FIREBASE_TOKEN:
      return {
        ...state,
        firebaseToken: action.token
      };
    default:
      return state;
  }
};
