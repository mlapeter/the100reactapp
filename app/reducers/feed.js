import {
  FETCH_FEED,
  FETCH_FEED_RESULT,
  FETCH_FEED_ERROR,
  FETCH_FEED_NO_DATA,
  LOAD_MORE_FEED_ITEMS,
  LOAD_MORE_FEED_ITEMS_RESULT,
  CHANGE_FEED_PAGE
} from "../actions/feed";

const initialState = {
  isLoading: false,
  feedItems: [],
  feedPage: 1,
  moreFeedItemsAvailable: true
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_FEED:
      return {
        ...state,
        isLoading: true,
        moreFeedItemsAvailable: true
      };
    case FETCH_FEED_RESULT:
      return {
        ...state,
        feedItems: action.result,
        isLoading: false
      };
    case FETCH_FEED_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false
      };
    case FETCH_FEED_NO_DATA:
      return {
        ...state,
        isLoading: false,
        moreFeedItemsAvailable: false
      };
    case LOAD_MORE_FEED_ITEMS:
      return {
        ...state,
        isLoading: true
      };
    case LOAD_MORE_FEED_ITEMS_RESULT:
      return {
        ...state,
        feedItems: [...state.feedItems, ...action.result],
        isLoading: false
      };

    case CHANGE_FEED_PAGE:
      return {
        ...state,
        feedPage: action.page
      };
    default:
      return state;
  }
};
