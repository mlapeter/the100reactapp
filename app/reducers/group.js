import {
  FETCH_GROUP,
  FETCH_GROUP_RESULT,
  FETCH_GROUP_ERROR,
  FETCH_GROUP_EMPTY,
  CHANGE_GROUP
} from "../actions/group";

const initialState = {
  isLoading: true,
  group: null,
  selectedGroupId: null
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_GROUP:
      return {
        ...state,
        isLoading: true,
        selectedGroupId: action.groupId
      };
    case FETCH_GROUP_RESULT:
      return {
        ...state,
        group: action.result,
        isLoading: false
      };
    case FETCH_GROUP_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false
      };
    case FETCH_GROUP_EMPTY:
      return {
        ...state,
        isLoading: false
      };
    case CHANGE_GROUP:
      return {
        ...state,
        selectedGroupId: action.groupId
      };
    default:
      return state;
  }
};
