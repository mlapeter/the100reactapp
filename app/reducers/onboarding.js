import {
  SET_PLATFORM,
  SET_GAMERTAG,
  SET_GAME,
  SET_PROFILE_INFO,
  CREATE_USER,
  CREATE_USER_ERROR
} from "../actions/onboarding";

const initialState = {
  age: "",
  email: "",
  gamertag: "",
  password: "",
  platform: "",
  timezone: "",
  play_schedule: "",
  play_style: "",
  group: "",
  primary_game_id: "",
  tos_privacy_agreement: false
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_PLATFORM:
      return {
        ...state,
        platform: action.platform
      };
      break;
    case SET_GAMERTAG:
      return {
        ...state,
        gamertag: action.gamertag
      };
      break;
    case SET_GAME:
      return {
        ...state,
        primary_game_id: action.gameId
      };
      break;
    case SET_PROFILE_INFO:
      return {
        ...state,
        age: action.age,
        play_schedule: action.playSchedule,
        play_style: action.playStyle,
        time_zone: action.timezone,
        group: action.group
      };
      break;
    case CREATE_USER:
      return {
        ...state,
        email: action.email,
        password: action.password,
        tos_privacy_agreement: action.tos_privacy_agreement
      };
      break;
    case CREATE_USER_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        // platform: "",
        // gamertag: "",
        // timezone: "",
        // play_style: "",
        // age: "",
        // play_schedule: "",
        // group: "",
        // email: "",
        password: ""
      };
    default:
      return state;
  }
};
