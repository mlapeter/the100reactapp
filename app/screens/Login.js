import React, { Component } from "react";
import {
  ActivityIndicator,
  Alert,
  AsyncStorage,
  Button,
  Dimensions,
  Image,
  Keyboard,
  KeyboardAvoidingView,
  View,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  TouchableOpacity,
  TouchableWithoutFeedback
} from "react-native";
// import { Analytics, PageHit } from "expo-analytics";
import PreSplash from "../components/PreSplash/PreSplash";
import MenuDrawer from "../router/index";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import { connect } from "react-redux";
import { connectAlert } from "../components/Alert";
import { fetchToken } from "../actions/authentication";
import { decodeToken } from "../actions/authentication";
import { setFirebaseToken } from "../actions/authentication";

import { colors, fontSizes, fontStyles } from "../styles";
const { height, width } = Dimensions.get("window");

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: null,
      password: null
    };
  }

  async saveItem(item, selectedValue) {
    try {
      await AsyncStorage.setItem(item, selectedValue);
    } catch (error) {
      console.error("AsyncStorage error: " + error.message);
    }
  }

  componentWillMount() {
    // const analytics = new Analytics(Environment["GOOGLE_ANALYTICS_ID"]);
    // analytics.hit(new PageHit("App Login"));
    // if (this.props.authentication.isAuthed === true) {
    //   this.props.navigation.navigate("Main");
    // }
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.authentication.error &&
      nextProps.authentication.errorAt !== this.props.authentication.errorAt
    ) {
      this.props.alertWithType(
        "error",
        "Error",
        nextProps.authentication.error
      );
    }
    if (nextProps.users.currentUser && nextProps.authentication.isAuthed) {
      this.props.navigation.navigate("App");
    }
  }

  // setTimeout(() => {
  //   if (
  //     this.props.authentication.isAuthed === true &&
  //     this.props.users.currentUser.gamertag != null
  //   ) {
  //     this.props.navigation.navigate("App");
  //   } else {
  //     this.props.navigation.navigate("Auth");
  //   }
  // }, 3000);

  userLogin() {
    if (!this.state.username || !this.state.password) return;
    this.props.dispatch(fetchToken(this.state.username, this.state.password));
    this.setState({
      username: "",
      password: ""
    });
  }

  render() {
    return (
      <KeyboardAwareScrollView
        contentContainerStyle={styles.container}
        extraHeight={10}
        keyboardOpeningTime={10}
      >
        <View style={styles.container}>
          <Image
            style={styles.image}
            source={require("../assets/images/logo.png")}
          />

          <TextInput
            onChangeText={username => this.setState({ username })}
            placeholder="Username"
            ref="username"
            textContentType="username"
            returnKeyType="next"
            style={styles.input}
            value={this.state.username}
            underlineColorAndroid={"transparent"}
          />

          <TextInput
            onChangeText={password => this.setState({ password })}
            placeholder="Password"
            ref="password"
            textContentType="password"
            secureTextEntry={true}
            style={styles.input}
            value={this.state.password}
            underlineColorAndroid={"transparent"}
          />

          {this.props.authentication.isLoading ? (
            <Text>Loading...</Text>
          ) : (
            <TouchableOpacity style={{}} onPress={this.userLogin.bind(this)}>
              <Text
                style={{
                  textAlign: "center",
                  padding: 15,
                  fontFamily: fontStyles.primaryFont,
                  fontSize: fontSizes.primary,
                  height: 80,
                  width: 300,
                  marginBottom: 15
                }}
              >
                Login
              </Text>
            </TouchableOpacity>
          )}
        </View>
      </KeyboardAwareScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 50,
    paddingBottom: 40
  },
  slogan: {
    fontSize: 40,
    margin: 20,
    textAlign: "center"
  },
  image: {
    padding: 5,
    resizeMode: "contain",
    height: 150 //height * 0.4 > 300 ? 300 : height * 0.4
  },
  loginContainer: {
    padding: 30,
    alignItems: "center",
    flex: 1
  },
  assuranceText: {
    textAlign: "center"
  },
  buttonStyle: {
    fontFamily: fontStyles.primaryFont,
    fontSize: fontSizes.primary,
    height: 100,
    width: 200,
    backgroundColor: colors.grey,
    marginBottom: 15
  },
  input: {
    margin: 15,
    padding: 5,
    fontFamily: fontStyles.primaryFont,
    fontSize: fontSizes.primary,
    borderWidth: 0.5,
    borderColor: "#d6d7da",
    height: 50,
    width: 300,
    flexDirection: "row"
  }
});

const mapStateToProps = state => {
  const authentication = state.authentication;
  const users = state.users;

  return {
    authentication,
    users
  };
};

export default connect(mapStateToProps)(connectAlert(Login));
