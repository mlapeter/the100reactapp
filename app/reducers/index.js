import { combineReducers } from "redux";

import authentication from "./authentication";
import search from "./search";
import notifications from "./notifications";
import users from "./users";
import group from "./group";
import gamingSessions from "./gamingSessions";
import onboarding from "./onboarding";
import conversations from "./conversations";
import feed from "./feed";

export default combineReducers({
  authentication,
  search,
  notifications,
  users,
  group,
  gamingSessions,
  onboarding,
  conversations,
  feed
});
