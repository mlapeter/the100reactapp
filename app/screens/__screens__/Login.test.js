import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import Login from "../Login";
import { initialState as authenticationInitialState } from "../../reducers/authentication";

const mockStore = configureStore([]);

it("successfully renders", () => {
  // const navigation = { state: { params: {} } };
  const initialState = { authentication: authenticationInitialState };
  const rendered = shallow(<Login />, {
    context: { store: mockStore(initialState) }
  });
  expect(rendered.dive().dive()).toMatchSnapshot();
});
