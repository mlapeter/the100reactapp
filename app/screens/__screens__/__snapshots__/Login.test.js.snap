// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfully renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    authentication={
      Object {
        "appLoading": true,
        "firebaseToken": "",
        "isAuthed": false,
        "isLoading": false,
        "token": "",
        "user": Object {},
      }
    }
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#23272A",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 150,
              "padding": 5,
              "resizeMode": "contain",
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          maxLength={100}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Username"
          placeholderStyle={
            Object {
              "color": "#2C2F33",
            }
          }
          returnKeyType="next"
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#43454a",
              "borderRadius": 2,
              "borderWidth": 0.5,
              "color": "#23272A",
              "flexDirection": "row",
              "fontSize": 20,
              "height": 50,
              "margin": 15,
              "padding": 5,
              "width": 300,
            }
          }
          textContentType="username"
          underlineColorAndroid="transparent"
          value={null}
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          maxLength={100}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderStyle={
            Object {
              "color": "#2C2F33",
            }
          }
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#43454a",
              "borderRadius": 2,
              "borderWidth": 0.5,
              "color": "#23272A",
              "flexDirection": "row",
              "fontSize": 20,
              "height": 50,
              "margin": 15,
              "padding": 5,
              "width": 300,
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
          value={null}
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "#4A90E2",
              "borderRadius": 3,
              "marginTop": 20,
              "paddingVertical": 20,
              "width": 300,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "textAlign": "center",
              }
            }
          >
            LOG IN
          </Text>
        </TouchableOpacity>
      </Component>,
      "contentContainerStyle": Object {
        "alignItems": "center",
        "backgroundColor": "#23272A",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingHorizontal": 5,
        "paddingVertical": 10,
        "width": "100%",
      },
      "enableAutoAutomaticScroll": true,
      "enableResetScrollToCoords": true,
      "extraHeight": 75,
      "extraScrollHeight": 0,
      "keyboardOpeningTime": 250,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Object {
                "height": 150,
                "padding": 5,
                "resizeMode": "contain",
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            maxLength={100}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Username"
            placeholderStyle={
              Object {
                "color": "#2C2F33",
              }
            }
            returnKeyType="next"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#43454a",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "color": "#23272A",
                "flexDirection": "row",
                "fontSize": 20,
                "height": 50,
                "margin": 15,
                "padding": 5,
                "width": 300,
              }
            }
            textContentType="username"
            underlineColorAndroid="transparent"
            value={null}
          />,
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            maxLength={100}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderStyle={
              Object {
                "color": "#2C2F33",
              }
            }
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#43454a",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "color": "#23272A",
                "flexDirection": "row",
                "fontSize": 20,
                "height": 50,
                "margin": 15,
                "padding": 5,
                "width": 300,
              }
            }
            textContentType="password"
            underlineColorAndroid="transparent"
            value={null}
          />,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#4A90E2",
                "borderRadius": 3,
                "marginTop": 20,
                "paddingVertical": 20,
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "textAlign": "center",
                }
              }
            >
              LOG IN
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#23272A",
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingHorizontal": 5,
          "paddingVertical": 10,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "height": 150,
              "padding": 5,
              "resizeMode": "contain",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCapitalize": "none",
            "maxLength": 100,
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Username",
            "placeholderStyle": Object {
              "color": "#2C2F33",
            },
            "returnKeyType": "next",
            "style": Object {
              "backgroundColor": "white",
              "borderColor": "#43454a",
              "borderRadius": 2,
              "borderWidth": 0.5,
              "color": "#23272A",
              "flexDirection": "row",
              "fontSize": 20,
              "height": 50,
              "margin": 15,
              "padding": 5,
              "width": 300,
            },
            "textContentType": "username",
            "underlineColorAndroid": "transparent",
            "value": null,
          },
          "ref": "username",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCapitalize": "none",
            "maxLength": 100,
            "onChangeText": [Function],
            "placeholder": "Password",
            "placeholderStyle": Object {
              "color": "#2C2F33",
            },
            "secureTextEntry": true,
            "style": Object {
              "backgroundColor": "white",
              "borderColor": "#43454a",
              "borderRadius": 2,
              "borderWidth": 0.5,
              "color": "#23272A",
              "flexDirection": "row",
              "fontSize": 20,
              "height": 50,
              "margin": 15,
              "padding": 5,
              "width": 300,
            },
            "textContentType": "password",
            "underlineColorAndroid": "transparent",
            "value": null,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "textAlign": "center",
                }
              }
            >
              LOG IN
            </Text>,
            "onPress": [Function],
            "style": Object {
              "backgroundColor": "#4A90E2",
              "borderRadius": 3,
              "marginTop": 20,
              "paddingVertical": 20,
              "width": 300,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "LOG IN",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "LOG IN",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#23272A",
              "flex": 1,
              "justifyContent": "flex-start",
              "paddingHorizontal": 5,
              "paddingVertical": 10,
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 150,
                "padding": 5,
                "resizeMode": "contain",
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            maxLength={100}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Username"
            placeholderStyle={
              Object {
                "color": "#2C2F33",
              }
            }
            returnKeyType="next"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#43454a",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "color": "#23272A",
                "flexDirection": "row",
                "fontSize": 20,
                "height": 50,
                "margin": 15,
                "padding": 5,
                "width": 300,
              }
            }
            textContentType="username"
            underlineColorAndroid="transparent"
            value={null}
          />
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            maxLength={100}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderStyle={
              Object {
                "color": "#2C2F33",
              }
            }
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#43454a",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "color": "#23272A",
                "flexDirection": "row",
                "fontSize": 20,
                "height": 50,
                "margin": 15,
                "padding": 5,
                "width": 300,
              }
            }
            textContentType="password"
            underlineColorAndroid="transparent"
            value={null}
          />
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#4A90E2",
                "borderRadius": 3,
                "marginTop": 20,
                "paddingVertical": 20,
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "textAlign": "center",
                }
              }
            >
              LOG IN
            </Text>
          </TouchableOpacity>
        </Component>,
        "contentContainerStyle": Object {
          "alignItems": "center",
          "backgroundColor": "#23272A",
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingHorizontal": 5,
          "paddingVertical": 10,
          "width": "100%",
        },
        "enableAutoAutomaticScroll": true,
        "enableResetScrollToCoords": true,
        "extraHeight": 75,
        "extraScrollHeight": 0,
        "keyboardOpeningTime": 250,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              source={1}
              style={
                Object {
                  "height": 150,
                  "padding": 5,
                  "resizeMode": "contain",
                }
              }
            />,
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              maxLength={100}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Username"
              placeholderStyle={
                Object {
                  "color": "#2C2F33",
                }
              }
              returnKeyType="next"
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#43454a",
                  "borderRadius": 2,
                  "borderWidth": 0.5,
                  "color": "#23272A",
                  "flexDirection": "row",
                  "fontSize": 20,
                  "height": 50,
                  "margin": 15,
                  "padding": 5,
                  "width": 300,
                }
              }
              textContentType="username"
              underlineColorAndroid="transparent"
              value={null}
            />,
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              maxLength={100}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderStyle={
                Object {
                  "color": "#2C2F33",
                }
              }
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#43454a",
                  "borderRadius": 2,
                  "borderWidth": 0.5,
                  "color": "#23272A",
                  "flexDirection": "row",
                  "fontSize": 20,
                  "height": 50,
                  "margin": 15,
                  "padding": 5,
                  "width": 300,
                }
              }
              textContentType="password"
              underlineColorAndroid="transparent"
              value={null}
            />,
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "#4A90E2",
                  "borderRadius": 3,
                  "marginTop": 20,
                  "paddingVertical": 20,
                  "width": 300,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "textAlign": "center",
                  }
                }
              >
                LOG IN
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#23272A",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Object {
                "height": 150,
                "padding": 5,
                "resizeMode": "contain",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCapitalize": "none",
              "maxLength": 100,
              "onChangeText": [Function],
              "onSubmitEditing": [Function],
              "placeholder": "Username",
              "placeholderStyle": Object {
                "color": "#2C2F33",
              },
              "returnKeyType": "next",
              "style": Object {
                "backgroundColor": "white",
                "borderColor": "#43454a",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "color": "#23272A",
                "flexDirection": "row",
                "fontSize": 20,
                "height": 50,
                "margin": 15,
                "padding": 5,
                "width": 300,
              },
              "textContentType": "username",
              "underlineColorAndroid": "transparent",
              "value": null,
            },
            "ref": "username",
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCapitalize": "none",
              "maxLength": 100,
              "onChangeText": [Function],
              "placeholder": "Password",
              "placeholderStyle": Object {
                "color": "#2C2F33",
              },
              "secureTextEntry": true,
              "style": Object {
                "backgroundColor": "white",
                "borderColor": "#43454a",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "color": "#23272A",
                "flexDirection": "row",
                "fontSize": 20,
                "height": 50,
                "margin": 15,
                "padding": 5,
                "width": 300,
              },
              "textContentType": "password",
              "underlineColorAndroid": "transparent",
              "value": null,
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "textAlign": "center",
                  }
                }
              >
                LOG IN
              </Text>,
              "onPress": [Function],
              "style": Object {
                "backgroundColor": "#4A90E2",
                "borderRadius": 3,
                "marginTop": 20,
                "paddingVertical": 20,
                "width": 300,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "LOG IN",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "LOG IN",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
